src/schemas/TimesheetSchema.tsx

```diff
@@ -1,12 +1,31 @@
 import { z } from "zod";
+import {RowSchema} from './RowSchema'; 
+
+// The status is either undefined, for not being at that stage yet, or 
+// contains the date and author of approving this submission 
+export const StatusEntryType = z.union(
+  [
+  z.object({
+    Date: z.number(), 
+    Author: z.string()
+  }), 
+  z.undefined()]); 
+
+// Status type contains the four stages of the pipeline we have defined 
+export const StatusType = z.object({
```

Comment:
I'm good with having one status object that has a different attribute for each stage of the process rather than a list of status entries, if that's easier for frontend display purposes. 

Can you just document your proposed schema on the diagram (scroll to the bottom, you'll see my proposed schema for the status schema stuff): https://lucid.app/lucidchart/912e6933-6883-4402-b926-805c934ded0c/edit?viewport_loc=-479%2C55%2C2998%2C1577%2C0_0&invitationId=inv_4b8281c9-b632-4438-9daa-06e3716331c5

Reply:
I think either would be fine, its just a matter of preference between iterating through a list of items / checking to see if certain things exist or just directly referencing the relevant fields to determine the current state (i.e. if field X / Y are not undefined, do Z, rather than just iterating through the list and then having like flags mutated to be true / false or something based on if a specific entry exists - this should be done on the backend anyways for validation purposes / to prevent someone from jumping stages illegally so whatever way you prefer for the backend we can go with

Reply:
Let's go with the status object with timestamps/undefined for each stage